### ISSUE TITLE
torch-tensorrt does not exist for ARM64, so Docker image creation fails

### ISSUE THREAD

**Issue by JonahMMay** [Jonah-May-OSS/wyoming-glados#126]

**Comment by NoDiskNoFun** [comment]
hey, i am using your whisper container on arm64 with dGPU and i wonder, as this issue is open for a while, did you ever fixed the problem? would be nice to have tts gpu accelerated too ...

**Comment by JonahMMay** [comment]
I don’t have an arm device with a dGPU to test/work on it with so I haven’t been able to make any progress myself.





Thank You,


Jonah May
________________________________

**Comment by JonahMMay** [comment]
@NoDiskNoFun It looks like torch-tensorrt may now have an ARM64 package. I added ARM64 images back to the release pipeline and the latest version is uploaded to Docker Hub. Can you give it a try?

**Comment by NoDiskNoFun** [comment]
[logs.txt](https://github.com/user-attachments/files/28826628/logs.txt)
You are awesome! :) I did pull the new image but it crashes on connect. logs are attached.

**Comment by NoDiskNoFun** [comment]
Oh wait, i had to wait a little longer. Now i am able to connect from Home Assistant.

2026-06-11 06:54:38.332345600 [INFO] gladostts.glados: Vocoder engine ready. TRT=True
2026-06-11 06:54:38.332550656 [INFO] gladostts.glados: Priming TRT engines with a minimal dummy run…
2026-06-11 06:54:40.704052736 [WARNING] py.warnings: /usr/src/wyoming-glados/gladostts/glados.py:315: UserWarning: RNN module weights are not part of single contiguous chunk of memory. This means they need to be compacted at every call, possibly greatly increasing memory usage. To compact weights again call flatten_parameters(). (Triggered internally at /pytorch/aten/src/ATen/native/cudnn/RNN.cpp:1479.)
  self._ensure_tacotron_jit().generate_jit(x, self.emb, alpha),

2026-06-11 06:54:40.704266240 [INFO] gladostts.glados: Warming up with buckets: [8, 16, 32]
2026-06-11 06:54:51.198735104 [INFO] gladostts.glados: Model warm-ups complete.
2026-06-11 06:54:51.946173440 [INFO] __main__: Server started and listening on tcp://0.0.0.0:10201

But the option is grayed out ...

<img width="375" height="196" alt="Image" src="https://github.com/user-attachments/assets/b0a55895-0713-4084-8e84-553aac658788" />

**Comment by NoDiskNoFun** [comment]
As an addition, starting the container did use some memory on my GPU.

<img width="1636" height="761" alt="Image" src="https://github.com/user-attachments/assets/11e9e662-9b24-4f92-9e12-63f1c6360346" />

**Comment by NoDiskNoFun** [comment]
Ah its greyed out because it can only do english language. With that said, i can confirm this container is fully working

**Comment by JonahMMay** [comment]
That's great to hear, and yes, it only does English now. I'll investigate at some point in the future to see if maybe it can do other languages, but since the models are only trained on English lines, I'm not sure it will be able to.

And yes, it does use some VRAM at first launch because it converts to standard TorchScript models to TensorRT. It isn't really something that can be done once and done because newer TensorRT updates tend to add enhancements that make the old models incompatible so it tries to use a model if it exists on disk, then re-converts it if it fails.

**Comment by NoDiskNoFun** [comment]
At least i found a german GLaDOS Voice, but its in the wrong format -> https://huggingface.co/systemofapwne/piper-de-glados. According to sh!tGPT one cannot convert it to pytorch .... Anyways, thanks for picking up this old ticket :)

**Comment by JonahMMay** [comment]
I'll have a new branch you can test out shortly. I can swap from an English-only phonemizer to a multi-lingual deep_phonemizer model, though accuracy is going to be questionable. We're still going to be using English-based training sets, so phonetic sounds that exist in other languages may not translate well. Maybe longer term we can figure out new training sets that incorporate audio from the other language options available in the games.

**Comment by JonahMMay** [comment]
Can you try out the branch [feat/multilanguage-path-a](https://github.com/Jonah-May-OSS/wyoming-glados/tree/feat/multilanguage-path-a)?
